src/Periphery/LiFiDEXAggregator.sol

```diff
@@ -732,6 +748,151 @@ contract LiFiDEXAggregator is WithdrawablePeriphery {
     }
 
+    /// @notice Performs a swap through iZiSwap V3 pools
+    /// @dev This function handles both X to Y and Y to X swaps through iZiSwap V3 pools
+    /// @param stream [pool, direction, to]
+    /// @param from Where to take liquidity for swap
+    /// @param tokenIn Input token
+    /// @param amountIn Amount of tokenIn to take for swap
+    function swapIzumiV3(
+        uint256 stream,
+        address from,
+        address tokenIn,
+        uint256 amountIn
+    ) private {
+        address pool = stream.readAddress();
+        uint8 direction = stream.readUint8(); // 0 = Y2X, 1 = X2Y
+        address to = stream.readAddress();
+
+        if (from == msg.sender) {
+            IERC20(tokenIn).safeTransferFrom(
+                msg.sender,
+                address(this),
+                amountIn
+            );
+        }
+
+        lastCalledPool = pool;
+
+        if (direction == DIRECTION_TOKEN0_TO_TOKEN1) {
+            IiZiSwapPool(pool).swapX2Y(
+                to,
+                uint128(amountIn),
+                IZUMI_LEFT_MOST_PT,
+                abi.encode(tokenIn)
+            );
+        } else {
+            IiZiSwapPool(pool).swapY2X(
+                to,
+                uint128(amountIn),
+                IZUMI_RIGHT_MOST_PT,
+                abi.encode(tokenIn)
+            );
+        }
+
+        if (lastCalledPool != IMPOSSIBLE_POOL_ADDRESS) {
+            revert IzumiV3SwapUnexpected();
+        }
+    }
+
+    /// @notice Performs a swap through SyncSwap pools
+    /// @dev This function handles both X to Y and Y to X swaps through SyncSwap pools.
+    ///      See [SyncSwap API documentation](https://docs.syncswap.xyz/api-documentation) for protocol details.
+    /// @param stream [pool, to, withdrawMode, isV1Pool, vault]
+    /// @param from Where to take liquidity for swap
+    /// @param tokenIn Input token
+    /// @param amountIn Amount of tokenIn to take for swap
+    function swapSyncSwap(
+        uint256 stream,
+        address from,
+        address tokenIn,
+        uint256 amountIn
+    ) private {
+        address pool = stream.readAddress();
+        address to = stream.readAddress();
+        if (pool == address(0) || to == address(0)) revert InvalidCallData();
+
+        // withdrawMode meaning for SyncSwap via vault:
+        //   1: Withdraw raw ETH (native)
+        //   2: Withdraw WETH (wrapped)
+        //   0: Let the vault decide (ETH for native, WETH for wrapped)
+        // For ERC-20 tokens the vault just withdraws the ERC-20
+        // and this mode byte is read and ignored by the ERC-20 branch.
+        uint8 withdrawMode = stream.readUint8();
+        bool isV1Pool = stream.readUint8() == 1;
+        if (isV1Pool) {
+            address vault = stream.readAddress();
+            if (vault == address(0)) revert InvalidCallData();
+            if (from == msg.sender) {
+                IERC20(tokenIn).safeTransferFrom(msg.sender, vault, amountIn);
```

Comment:
Generally we should be using our LibAsset transfer function here but since we will have to switch to an older LibAsset version to be able to deploy to zksync, I think it doesnt make sense to make that switch. 
However, I will leave this comment for @mirooon to consider this in future versions.

Reply:
+1